pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManagerStatelessTest.java

```diff
@@ -163,6 +163,8 @@ public void testPinotTaskManagerSchedulerWithRestart()
     // Restart controller
     stopController();
     startController(properties);
+    // wait for controller to start correctly.
+    Thread.sleep(1000);
```

Comment:
Do you run into issues without this wait? We should avoid using this kind of wait to get instance ready as this can be flaky

Reply:
yeah. for some unknown reason this test will fail if I dont wait. i can do a "TestUtils.waitUntilCondition" instead.

Reply:
Let's check what is causing the flakiness, and use `TestUtils.waitUntilCondition` to guard that condition

Reply:
i think we just need to make sure the HTTPServer is up . see my change

Reply:
Why is the HTTP server still not ready when the start controller already returns? IMO it should be ready during the start

Reply:
this I have no idea. the error simply said
```
Mar 28, 2022 6:31:51 PM org.glassfish.grizzly.http.server.NetworkListener shutdownNow
INFO: Stopped listener bound to [0.0.0.0:18998]
Mar 28, 2022 6:31:52 PM org.glassfish.grizzly.http.server.NetworkListener start
INFO: Started listener bound to [0.0.0.0:18998]
Mar 28, 2022 6:31:52 PM org.glassfish.grizzly.http.server.HttpServer start
INFO: [HttpServer-1] Started.

org.apache.http.NoHttpResponseException: localhost:18998 failed to respond

	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:141)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:157)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
...
	at org.apache.pinot.common.utils.http.HttpClient.sendJsonPutRequest(HttpClient.java:257)
	at org.apache.pinot.controller.helix.ControllerRequestClient.updateTableConfig(ControllerRequestClient.java:100)
	at org.apache.pinot.controller.helix.ControllerTest.updateTableConfig(ControllerTest.java:540)
	at org.apache.pinot.controller.helix.core.minion.PinotTaskManagerStatelessTest.testPinotTaskManagerSchedulerWithRestart(PinotTaskManagerStatelessTest.java:180)
```

Reply:
@walterddr can you create an issue to track the above ? I think we can merge this

Reply:
done see #8431